tools/SWTbot-Testing/src/swtbot_testing/MavenSwtest.java

```diff
@@ -0,0 +1,453 @@
+
+package swtbot_testing;
+
+import static org.junit.Assert.*;
+
+import java.awt.List;
+import java.util.ArrayList;
+
+import javax.swing.text.html.HTMLDocument.Iterator;
+
+import org.eclipse.swtbot.eclipse.finder.SWTWorkbenchBot;
+import org.junit.AfterClass;
+import org.junit.BeforeClass;
+import org.junit.Ignore;
+import org.junit.Test;
+
+public class MavenSwtest
+{
+
+    private static SWTWorkbenchBot bot;
+
+    @BeforeClass
+    public static void beforeClass()
+    {
+        bot = new SWTWorkbenchBot();
+
+        bot.viewByTitle( "Welcome" ).close();
+        bot.toolbarButtonWithTooltip( "Open Perspective" ).click();
+        bot.table().select( "Liferay Plugins" );
+        bot.button( "OK" ).click();
+
+        bot.toolbarDropDownButtonWithTooltip( "Create a new Liferay Plugin Project" ).menuItem(
+            "New Liferay Plugin Project" ).click();
+    }
+
+    public boolean matchItemInItems( String[] actualItems, String expectedItems )
+    {
+
+        for( String item : actualItems )
+        {
+            if( item.equals( expectedItems ) )
+                return true;
+        }
+        return false;
+    }
+
+    @Test
+    public void newMavenPortletProjectWithCreateMavenProfile()
+    {
+
+        /* �鿴ҳ��ĳ�ʼ״̬ */
+
+        String validationMessage = bot.text( 2 ).getText();
+        assertEquals( "Please enter a project name.", validationMessage );// У����ʾ
+        assertEquals( "", bot.textWithLabel( "Project name:" ).getText() );
+        assertEquals( "", bot.textWithLabel( "Display name:" ).getText() );
+
+        /* ���Build type�� Ant (liferay-plugins-sdk) */
+
+        if( ( bot.comboBoxWithLabel( "Build type:" ).getText() ).equals( "Ant (liferay-plugins-sdk)" ) )
+        {
+
+            assertEquals( "Ant (liferay-plugins-sdk)", bot.comboBoxWithLabel( "Build type:" ).getText() );
+            assertEquals( "Portlet", bot.comboBoxWithLabel( "Plugin type:" ).getText() );
+
+            String expected_buildtype[] = { "Ant (liferay-plugins-sdk)", "Maven (liferay-maven-plugin)" };
+            String expected_plugintype[] =
+                { "Portlet", "Service Builder Portlet", "Hook", "Layout Template", "Theme", "Ext", "Web" };
+            String actual_buildtype[] = bot.comboBoxWithLabel( "Build type:" ).items();
+            String actual_plugintype[] = bot.comboBoxWithLabel( "Plugin type:" ).items();
+
+            assertEquals( expected_plugintype.length, actual_plugintype.length );
+            for( String expected_b : expected_buildtype )
```

Comment:
Notes: it's better to start in a new line when adding a methond, as well as "for(){}"